proposals/NNNN-rpc-message-protection.md

```diff
@@ -0,0 +1,215 @@
+    </param>
+</function>
+
+<struct name="HMIPermissions" since="2.0">
+    <param name="allowed" type="HMILevel" minsize="0" maxsize="100" array="true" mandatory="true">
+        <description>A set of all HMI levels that are permitted for this given RPC.</description>
+    </param>
+    <param name="userDisallowed" type="HMILevel" minsize="0" maxsize="100" array="true" mandatory="true">
+        <description>A set of all HMI levels that are prohibited for this given RPC.</description>
+    </param>
+</struct>
+
+<struct name="ParameterPermissions" since="2.0">
+    <param name="allowed" type="String" minsize="0" maxsize="100" maxlength = "100" array="true" mandatory="true">
+        <description>A set of all parameters that are permitted for this given RPC.</description>
+    </param>
+    <param name="userDisallowed" type="String" minsize="0" maxsize="100" maxlength = "100" array="true" mandatory="true">
+        <description>A set of all parameters that are prohibited for this given RPC.</description>
+    </param>
+</struct>
+```
+
+### Mobile proxy change (Android and iOS)
+SDL proxy shall support sending a PRC with both encrypted and unencrypted format as it currently does if the RPC service has encryption enabled. 
+
+SDL proxy may restart service 7 with encryption enabled in several situations. 
+-	1. Before SDL proxy sends the first RPC message that need encryption. This works like delay loading of a dynamic library. Only when encryption is truly needed, proxy enables encryption for RPC service.  This is the latest time. 
+-	2. After SDL proxy receives an `OnPermissionsChange` notification and there is at least one RPC needs encryption. In this case, proxy enables encryption for any potential usage. This is earliest time that the app/proxy knows it may need encryption. 
+-	3. Sometime in between. For example, when the driver activates the app and HMI brings the app to foreground (SDL proxy receives `OnHMIStatus` notification with `hmiLevel=FULL`). Due to the fact that it may take a long time (up to a minute) to get RPC service enabled, and there is no overhead after that, we strongly recommend proxy enable encryption after the app get activated and there is at least one RPC need protection.
+
+
+SDL proxy shall only send encrypted RPC requests and receive encrypted RPC responses and notifications in an encryption enabled service if the corresponding RPC needs encryption.
+
+SDL proxy shall send/receive encrypted RPC requests/responses if the app chooses to `setPayloadProtected(true)` in an encryption enabled service even if the corresponding RPC does not need encryption.
+
+
+
+### SDL core change
+
+#### Proposed solution
+Before the encryption of RPC service 7 is enabled, SDL rejects any RPC request if the RPC needs protection. (Please see policy updates for which RPC needs protection.) SDL continues processing a RPC request if the RPC does not need protection. SDL sends a notification only if the RPC does not need protection.
```

Comment:
SDL sends a notification only if the **notification** RPC does not need protection.